ksql-engine/src/test/java/io/confluent/ksql/planner/plan/JoinNodeTest.java

```diff
@@ -186,15 +189,17 @@ protected int getPartitions(KafkaTopicClient kafkaTopicClient) {
 
     RightTable rightTable = new RightTable(new PlanNodeId("1"), joinNode.getRight().getSchema());
 
-    JoinNode testJoinNode = new JoinNode(joinNode.getId(), joinNode.getType(), joinNode.getLeft()
+    JoinNode testJoinNode = new JoinNode(joinNode.getId(), joinNode.getJoinType(), joinNode.getLeft()
         , rightTable, joinNode.getLeftKeyFieldName(), joinNode.getRightKeyFieldName(), joinNode
-                                             .getLeftAlias(), joinNode.getRightAlias());
+                                             .getLeftAlias(), joinNode.getRightAlias(), null,
+                                         DataSource.DataSourceType.KSTREAM,
+                                         DataSource.DataSourceType.KTABLE);
     testJoinNode.tableForJoin(builder, ksqlConfig, kafkaTopicClient, functionRegistry,
                           new HashMap<>(), new MockSchemaRegistryClient());
 
   }
 
-  @Test
+  @Test @Ignore
```

Comment:
Any reason for ignoring this test?

Reply:
Yes. It started to fail. I have to investigate, though I think the existing tests in this class are superseded be the new test cases I added using real mocks.

Reply:
The reason both the tests were failing was because in the old code, the table was built before the stream. The new code builds the stream before the table, causing the assumptions to fail.

Since the ordering really doesn't matter in any way, I just switched the order in which the stream and table are built in a stream-table left join. This causes the tests to pass as is.